semeio/jobs/rft/utility.py

```diff
@@ -0,0 +1,265 @@
+
+    return trajectory_points
+
+
+def load_and_parse_well_time_file(filename):
+    """
+    The file must be on the format of <STRING INT INT INT INT> which
+    represents <well_name day month year report_step>
+
+    Parameters
+    ----------
+    filename : string
+        Filename to open
+
+    Returns
+    -------
+    [Tuple]
+        Returns a list of tuples with (well, datetime, report_step)
+    """
+
+    if not os.path.isfile(filename):
+        raise argparse.ArgumentTypeError("The path {} does not exist".format(filename))
+
+    well_times = []
+    base_error_msg = (
+        "Line {line_number} in well_and_time_file {filename} not on proper format: "
+    )
+
+    with open(filename) as well_time_file:
+        lines = well_time_file.readlines()
+
+    well_time_lines = [(strip_comments(l), i + 1) for i, l in enumerate(lines)]
+
+    for line, line_number in well_time_lines:
+        tokens = line.split()
+
+        if not tokens:
+            continue
+
+        elif len(tokens) != 5:
+            err_msg = (
+                base_error_msg
+                + "Unexpected number of tokens: expected 5 got {num_tokens}"
+            )
+            raise argparse.ArgumentTypeError(
+                err_msg.format(
+                    line_number=line_number, filename=filename, num_tokens=len(tokens)
+                )
+            )
+
+        well = tokens[0]
+
+        try:
+            report_step = int(tokens[4])
+        except ValueError:
+            err_msg = base_error_msg + "Unable to convert {report_step} to int"
+            raise argparse.ArgumentTypeError(
+                err_msg.format(
+                    line_number=line_number, filename=filename, report_step=tokens[4]
+                )
+            )
+
+        try:
+            year = int(tokens[3])
+            month = int(tokens[2])
+            day = int(tokens[1])
+            time = datetime.date(year, month, day)
+
+        except ValueError:
+            err_msg = (
+                base_error_msg
+                + "Unable to parse date, expected day month year got: {day} {month} {year}"
+            )
+            raise argparse.ArgumentTypeError(
+                err_msg.format(
+                    line_number=line_number,
+                    filename=filename,
+                    day=tokens[1],
+                    month=tokens[2],
+                    year=tokens[3],
+                )
+            )
+
+        well_times.append((well, time, report_step))
+
+    return well_times
+
+
+def load_and_parse_zonemap_file(filename):
+    """
+    The file must be on the format of <INT STRING*> (any number of strings)
+    which represents <k_value zone_names..>
+
+    Parameters
+    ----------
+    filename : string
+        Filename to open
+
+    Returns
+    -------
+    Dict
+        Returns a dict with k_value as keys and a list of corresponding
+        zonenames
+    """
+
+    zones_at_k_value = {}
+
+    with open(filename, "r") as f:
```

Comment:
No test whether file exist...

Reply:
It is tested in the caller (`load_zonemap(path)`). The reason it is split is because the `load_zonemap` returns a `ZoneMap` object, while `load_and_parse_zonemap_file` returns the content of the zonemap file in a well defined data structure.

It could perhaps be one function though..

Reply:
This is refactored never the less, and they are now in the same location